libs/geography/src/drivers/magento/geography.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { Apollo } from 'apollo-angular';
+
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { DaffGeographyServiceInterface } from '../interfaces/geography-service.interface';
+import { DaffCountry } from '../../models/country';
+import { DaffMagentoCountryTransformer } from './transforms/responses/country.service';
+import { getCountries, MagentoGetCountriesResponse } from './queries/public_api';
+
+/**
+ * A service for making Magento GraphQL queries for carts.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffMagentoGeographyService implements DaffGeographyServiceInterface<DaffCountry> {
+  constructor(
+    private apollo: Apollo,
+    public countryTransformer: DaffMagentoCountryTransformer,
+  ) {}
+
+  list(): Observable<DaffCountry[]> {
+    return this.apollo.query<MagentoGetCountriesResponse>({
+      query: getCountries,
+    }).pipe(
+      map(result => result.data.countries
```

Comment:
Is `data.countries` potentially nullable or empty? Why the condition?

Reply:
I've been surprised in the past by `null` where I expected an empty array so I made a habit of checking all data coming from Magento to avoid null pointer errors. This specific instance is _probably_ safe to remove.

Reply:
Ok. Then what I would likely do is:

```
list(): Observable<DaffCountry[]> {
    return this.apollo.query<MagentoGetCountriesResponse>({
      query: getCountries,
    }).pipe(
      map(result => result.data.countries ? result.data.countries : []),
      map(result => result.data.countries.map(country => this.countryTransformer.transform(country))
      )
    );
  }
```

Reply:
How do you feel about `map(result => result.data.countries || [])` as a method of setting a default value?

Reply:
lgtm

Reply:
changed